src/Analyser/NodeScopeResolver.php

```diff
@@ -2111,17 +2112,15 @@ static function () use ($scope, $expr): MutatingScope {
 								if (count($directClassNames) === 1) {
 									$scopeClass = $directClassNames[0];
 									$thisType = new ObjectType($scopeClass);
-								} elseif (
-									$argValue instanceof Expr\ClassConstFetch
-									&& $argValue->name instanceof Node\Identifier
-									&& strtolower($argValue->name->name) === 'class'
-									&& $argValue->class instanceof Name
-								) {
-									$scopeClass = $scope->resolveName($argValue->class);
-									$thisType = new ObjectType($scopeClass);
 								} elseif ($argValueType instanceof ConstantStringType) {
 									$scopeClass = $argValueType->getValue();
 									$thisType = new ObjectType($scopeClass);
+								} elseif (
+									$argValueType instanceof GenericClassStringType
+									&& $argValueType->getGenericType() instanceof ObjectType
```

Comment:
instanceof TypeWithClassName would be better. Otherwise 👍 .

Reply:
done, 👍